src/utils/validateFilepath.ts

```diff
@@ -0,0 +1,17 @@
+import path from 'path';
+import * as config from '../config';
+import { Request, Response } from 'express';
+
+// https://codeql.github.com/codeql-query-help/javascript/js-path-injection/
+
+export const validateFilepath = (req: Request, res: Response): string | void => {
```

Comment:
I would have passed just the `req.path` and returned the normalised path. Otherwise, I would have thrown an error to delegate the logging of the error and the return to the catch section in the validation middleware, something like the following:
```suggestion
...
export const validateFilepath = (reqPath: string): string => {

    const normalisedPath = path.resolve(config.ROOT_URL, reqPath);

    if (normalisedPath.startsWith(config.ROOT_URL)) {
        return normalisedPath;
    }

    throw new Error("Validation file path failed!");
};

```

Reply:
Yep I agree, this simplify logics and test mocks.